metadata/src/main/java/org/apache/kafka/metadata/authorizer/StandardAuthorizer.java

```diff
@@ -97,17 +119,17 @@ public void completeInitialLoad(Exception e) {
 
     @Override
     public void addAcl(Uuid id, StandardAcl acl) {
-        data.addAcl(id, acl);
+        inWriteLock(() -> data.addAcl(id, acl));
```

Comment:
I am not sure how much it matters, but it would probably be more efficient for `addAcl` to be a batched API. Perhaps it is ok since hopefully most of the time the brunt of the initialization is in `loadSnapshot`.